### DISCUSSION TITLE
Uploaded files not being migrated correctly to other server

### DISCUSSION THREAD

**Discussion by teratux** [sulu/sulu#6384] [Q&A] [answered]
| Q | A
| --- | ---
| Sulu Version | 2.3.4
| PHP Version | 7.4.3
| DB Version | Postgresql 12
| Browser Version | Google Chrome 87.0.4280.66

# Actual Behavior

After deploying from a STAGE server to a PROD server using the deployer thumbnails of uploaded files and some pages are not being shown correctly on the new server.

# Expected Behavior

No content loss

# Steps to Reproduce

1- on the STAGE server: ./vendor/bin/dep deploy prod
2- on the PROD server: ./bin/console sulu:admin:update-build
3- rsync all files from the shared folder to PROD's shared folder
4- open the admin interface no thumnails shown
5- open some of the pages from the website and content is not being shown correctly, some pages empty, others without the uploaded files

on PROD:
![image](https://user-images.githubusercontent.com/92475/143730206-80b51e0b-44e6-4594-94b1-a072b1f80745.png)

on STAGE:
![image](https://user-images.githubusercontent.com/92475/143730214-59bfc9d3-d270-4e45-abdc-833cd1663579.png)

perhaps I'm not following a standard way of deploying uploaded files, the deployer did not manage uploaded content in my case and had to resort to manual operation

Thanks

**Comment by alexander-schranz** [comment, marked as answer]
Hello @teratux,

This seems not be an issue with sulu. More an issue how you did migrate your data.
The tool you mention `deployer` I only know from code deployments and not as usage to move a installation from one server to another server with its data.

Normally if you want to update your ServerA with data from ServerB you need todo the following:

1. Make a `database dump` of your installation on ServerB and import it into ServerA
2. Copy your uploaded files from `var/uploads` from ServerB to ServerA
3. Make sure that you are running both on the same version of your code (branch, commit, tagged version, ...)
4. Clear the caches (symfony, http caches, includes eventually third party proxies, cdn caches, do not forget eventually configured redis caches)
5. Purge and Reindex the Massive Search Data

Then all your page data in the admin should be there as it was on the other server. If your thumbnails don't work you should have a look at your browser network tab to find out more. In most cases its a false configured vhost or you did not define the permissions correctly after your rsync. Have for this a look at the server docs:

 - https://docs.sulu.io/en/latest/cookbook/web-server/apache.html
 - https://docs.sulu.io/en/latest/cookbook/web-server/nginx.html
 
Which will show you an example vhost and how to set the permissons for the directories correctly. I hope I could help you with this.

**Reply by teratux** [reply]
I finally decided to compress the entire site and copy it to PROD and continue to work on from there, and thumbnails are working as expected.  I tried rebuilding the index as you suggested, before I had completed steps 1 through 4 but still had issues with the thumbnails.

Thanks

**Reply by alexander-schranz** [reply]
Thumbnails are in sulu generated on request. So they don't need to be copied from one server to another. On the next request the image will be generated and stored into the public directory for fast access. Make sure that you try to upload a new image and that the thumbnail works also for that. If that does not work as said it mostly is an error on the server VHost configuration or a problem with Permissions.
